Alerting: Introduce proper feature toggles for common state history backend combinations (#65497)
grafana/grafana: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md

| `influxdbBackendMigration`         | Query InfluxDB InfluxQL without the proxy                                                                                                                                                           |
| `clientTokenRotation`              | Replaces the current in-request token rotation so that the client initiates the rotation                                                                                                            |
| `prometheusDataplane`              | Changes responses to from Prometheus to be compliant with the dataplane specification. In particular it sets the numeric Field.Name from 'Value' to the value of the `__name__` label when present. |

> | `alertStateHistoryLokiSecondary`   | Enable alert state history to be written to an external loki instance in addition to Grafana annotations.                                                                                           |

Author: Sure, right now alerting happens to (ab)use Grafana Annotations as a method of storing history for when alerts fire/resolve. It writes a lot of rows to the annotations table that aren't really associated with "lines on panels" which was the intended purpose of annotations.

We're moving the storage mechanism for this data off annotations to something more bespoke and giving alerting its own API for state history. The new storage has pluggable backends (including annotations, if OSS customers wish to keep it), but in grafana cloud, we'll be automatically configuring a Loki instance for users as the default store for this data. Queries will dispatch to whichever store is configured.

The idea is to have three stages of maturity here, for users who want to try enabling the new Loki backend:
- One where we write to loki in addition to the current SQL table (which only _happens_ to be the annotations table)
- One where Loki will be the primary backend for serving queries, but we still write the old annotations as a backup
- One where we allow a full switch-over to the external system

> | `alertStateHistoryLokiSecondary`   | Enable alert state history to be written to an external loki instance in addition to Grafana annotations.                                                                                           |

Reviewer: 👍

> | `alertStateHistoryLokiPrimary`     | Enable usage of a remote Loki instance as the primary source for state history reads.                                                                                                               |

Reviewer: ```suggestion
| `alertStateHistoryLokiPrimary`     | Enable a remote Loki instance to be the primary source for state history reads.                                                                                                               |
```

> | `alertStateHistoryLokiOnly`        | Allow Grafana annotations emitted by alerting to be disabled completely when remote Loki is in place.                                                                                               |

Reviewer: ```suggestion
| `alertStateHistoryLokiOnly`        | Disable Grafana alerts from emitting annotations when a remote Loki instance is available.                                                                                               |
```

Author: This is much better wording, great suggestion
